antonTan96 iP

antonTan96 opened the pull request:
### Cracker++

> Productivity is less about what you do with your time. And more about how you run your mind. [[Robin S. Sharma](https://quotefancy.com/quote/953359/Robin-S-Sharma-Productivity-is-less-about-what-you-do-with-your-time-And-more-about-how)]

Cracker++ frees your mind of having to remember things you need to do. It's

- Text-based
- Easy to use
- **SUPER FAST**

To use Cracker++, follow the steps below:

1. Download Cracker from [here (To be updated)](url)
2. double click it
3. add your tasks
4. You're done!

Features:

 

- [x] Managing tasks
- [x]  Managing deadlines
- [x]  Managing events
- [ ]  Displaying your list of tasks
- [ ]  Searching for a specific keyword


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public static void main(String[] args) {
        Cracker bot = new Cracker();


        bot.startService();


    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

antonTan96 committed:
Experimental push

antonTan96 committed:
Level-0
Implemeted skeletal version of Cracker(formally Duke)

antonTan96 committed:
Level-1
added echo function

antonTan96 committed:
Level-2
Added ability to store task and list task

antonTan96 committed:
Level-3
Added ability to mark tasks as done and undone

antonTan96 committed:
Level-4
Added types of Tasks

antonTan96 committed:
A-TextUiTesting
Fixed bug for automated testing

antonTan96 committed:
Level-5
Added error Handling

antonTan96 committed:
A-Enums
Added enums

antonTan96 committed:
Level-5
Added Exceptions handling

antonTan96 committed:
Level-6
Added ability to delete tasks

antonTan96 committed:
Add Save Feature

antonTan96 committed:
Merge branch 'branch-Level-7'

antonTan96 committed:
Add Dates and Times

antonTan96 committed:
Merge branch 'branch-Level-8'

antonTan96 committed:
Refactor code

antonTan96 committed:
Add Parser and Refactor code

antonTan96 committed:
Revert "Refactor code"

This reverts commit c83e2b32f4a70302f1695219417d7d187cf8c2cc.

antonTan96 committed:
Merge pull request #1 from antonTan96/add-gradle-support

Add gradle support

antonTan96 committed:
Add dependancies

antonTan96 committed:
Add JUnit Test

antonTan96 committed:
Add test cases

antonTan96 committed:
Add Testing

antonTan96 committed:
Stash changes

antonTan96 committed:
Fixed errors

antonTan96 committed:
Resolve merge conflict and Add Javadocs

antonTan96 committed:
Apply Coding Standard

antonTan96 committed:
Add Find function

antonTan96 committed:
Merge tag 'Level-9'

antonTan96 committed:
Merge tag 'A-CodingStandard'

antonTan96 committed:
Merge tag 'A-Javadoc'

ncmathan commented on src/main/java/crackerpackage/Cracker.java:
```diff
@@ -0,0 +1,128 @@
+    /**
+     * The types of operations supported by the chatbot.
+     */
+    public enum Type {
```
I like how you used enums to distinguish the various tasks to be done. Keep it up!

ncmathan commented on src/main/java/crackerpackage/Cracker.java:
```diff
@@ -0,0 +1,128 @@
+            } catch (IllegalFormatException e) {
+                reply.echo(e.toString());
+            }
+
```
Have you considered shifting the switch statement block to the parser, so that when new code formats needs to be parsed, only Parser needs to be modified? 🤔

ncmathan commented on src/main/java/crackerpackage/Storage.java:
```diff
@@ -0,0 +1,126 @@
+            } else if (task instanceof Deadline) {
+                taskString.append(LocalDate.parse(((Deadline) task).getDeadline(), formatter));
+                taskString.append('|');
+            }
```
You can consider using only one append statement at each section by concatenating the strings together to reduce code length.

ncmathan reviewed (commented):
Overall great attempt following the coding standards, great implementation of javadocs as well. Keep it up! 👍

zhonghan721 commented on src/main/java/crackerpackage/Cracker.java:
```diff
@@ -0,0 +1,128 @@
+    /**
+     * Starts up the bot.
+     */
+    public void startService() {
```
I like how you used startService as the name. It relates well to the concept of a chatbot.

zhonghan721 commented on src/main/java/crackerpackage/tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+ */
+
+public abstract class Task {
+    private boolean isDone;
```
Great that the boolean name sounds like a boolean 👍

zhonghan721 reviewed (commented):
Good job in the naming of the fields and methods! It was easy to figure out the purpose of each method/field.

antonTan96 committed:
Add GUI

antonTan96 committed:
Add Greeting dialog

antonTan96 committed:
Merge branch 'branch-Level-10'

antonTan96 committed:
Delete src/main/java/UIComponents directory

UIComponents is a package name, which causes errors when testing with gradle due to the capital letters contained.

Refactoring the directory will solve the problem

Let's relocate the files deleted to uicomponents

antonTan96 committed:
Refactor code

The files added were in the UIComponents package directory which was deleted

Having capital letters in packages cause problems with gradle testing.

The files in the deleted directory will be relocated to this directory.

Let's upload the files that were originally in UIComponents here.

antonTan96 committed:
Use java assertion feature

antonTan96 committed:
Refactor code

Storage currently handles the load function, loading new tasks given a
string.

Feeding the string to the parser would be more fitting of OOP
principles.

Let's refactor the code to have the parser handle task regeneration.

antonTan96 committed:
Fix coding standard

antonTan96 committed:
Merge pull request #3 from antonTan96/branch-A-Assertions

Use java assertion feature

antonTan96 committed:
Revert "Use java assertion feature"

antonTan96 committed:
Merge pull request #4 from antonTan96/revert-3-branch-A-Assertions

Revert "Use java assertion feature"

antonTan96 committed:
Revert "Revert "Use java assertion feature""

antonTan96 committed:
Merge pull request #5 from antonTan96/revert-4-revert-3-branch-A-Assertions

Revert "Revert "Use java assertion feature""

antonTan96 committed:
Merge pull request #2 from antonTan96/branch-A-CodeQuality

Improve code quality

antonTan96 committed:
Add Archive and clear task function

antonTan96 committed:
Add product Screenshot

antonTan96 committed:
Add find command

The find command was not added while refactoring files.

Let's add the find command back.

antonTan96 committed:
Update README
